fea-rs/src/compile/tables/var_store.rs

```diff
@@ -0,0 +1,312 @@
+//! Building the ItemVariationStore
+//!
+//! this will probably get moved to write-fonts at some point?
+
+use std::collections::HashMap;
+
+use indexmap::IndexMap;
+use write_fonts::tables::variations::{
+    ItemVariationData, ItemVariationStore, VariationRegion, VariationRegionList,
+};
+
+///
+/// this probably ends up living in write-fonts
+#[derive(Clone, Debug, Default)]
+pub(crate) struct VariationStoreBuilder {
+    // region -> index map
+    all_regions: HashMap<VariationRegion, usize>,
+    // we use an index map so that we have a determenistic ordering,
+    // which lets us write better tests
+    delta_sets: IndexMap<RegionSet, IndexMap<DeltaSet, DeltaKey>>,
+    next_id: DeltaKey,
+}
+
+#[derive(Clone, Debug, Default)]
+pub(crate) struct VariationStoreKeyLookup {
+    map: HashMap<DeltaKey, DeltaKey>,
+}
+
+/// a subset of regions, represented as indicies into the region map
+#[derive(Clone, Debug, Hash, PartialEq, Eq)]
+struct RegionSet {
+    indicies: Vec<u16>,
+}
+
+#[derive(Clone, Debug, Default, Hash, PartialEq, Eq)]
+struct DeltaSet {
+    values: Vec<i32>,
+}
```

Comment:
Not loving Region**Set** and Delta**Set** ... that are not actually sets. Could these be tuple structs with non-set naming, e.g. `struct RegionIndices(Vec<u16>)`?

Reply:
I think the "delta sets" name comes from the OT Variations spec

https://learn.microsoft.com/en-us/typography/opentype/spec/otvaroverview

> Delta set: A set of adjustment deltas associated with a particular region of the variation space.

Reply:
`ItemVariationData` actually contains an array of `DeltaSet`s:

https://learn.microsoft.com/en-us/typography/opentype/spec/otvarcommonformats

the name "delta-set" is also used to refer to the rows of the ItemVarStore's VariationData subtables, that contain deltas for each region referenced by that subtable.  I guess they might be thought of as a "set" in the sense that their ordering isn't essential and can/does change when optimizing the varstore.

Reply:
As cosimo mentions, 'delta set' is a term used in the spec, and the 'region set' *is* actually a set, but this is a moot point (this type is removed when I implement the optimizer in #188)